packages/opentelemetry-resources/src/Resource.ts

```diff
@@ -59,4 +59,11 @@ export interface Resource {
   merge(other: Resource | null): Resource;
 
   getRawAttributes(): RawResourceAttribute[];
+
+  /**
+   * Get the schema URL for this resource.
+   *
+   * @returns the schema URL or undefined if not set
+   */
+  getSchemaUrl?(): string | undefined;
```

Comment:
I think I'd prefer a readonly property, then use a getter in the `ResourceImpl` to enforce it,  like we do with the other properties to keep it consistent.

cc @dyladan do you have a preference? (since you re-did resources in the latest major version)

Reply:
made this change in https://github.com/open-telemetry/opentelemetry-js/pull/5753/commits/a62f67e06928276b05b8f9afe11a7bea184eb33e

can revert if preferred of course